src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -807,16 +808,64 @@ protected static int GetParameterIndex<TNode>(SeparatedSyntaxList<TNode> paramet
                             continue;
                         }
 
+                        SyntaxNode expression = null;
+
+                        var callsiteInferenceFailed = false;
+
+                        if (addedParameter.CallSiteKind == CallSiteKind.Inferred && addedParameter.TypeBinds && getSemanticModelAndRecommendations != null)
+                        {
+                            callsiteInferenceFailed = true;
+
+                            (var semanticModel, var recommendations) = await getSemanticModelAndRecommendations().ConfigureAwait(false);
+
+                            var targetType = addedParameter.Type;
+
+                            var sourceSymbols = recommendations.Where(r => r.IsNonImplicitAndFromSource());
+                            var sourceSymbolsWithTypes = sourceSymbols.Where(s => s.GetSymbolType() != null);
+                            var orderedSourceSymbolsWithTypes = sourceSymbolsWithTypes.OrderByDescending(s => s.Locations.First().SourceSpan.Start);
+
+                            var localsAndParameters = orderedSourceSymbolsWithTypes.Where(s => s.IsKind(SymbolKind.Local) || s.IsKind(SymbolKind.Parameter));
+                            var propertiesAndFields = orderedSourceSymbolsWithTypes.Where(s => s.IsKind(SymbolKind.Property) || s.IsKind(SymbolKind.Field));
+
+                            var fullyOrderedSymbolsByKind = localsAndParameters.Concat(propertiesAndFields);
+
+                            foreach (var symbol in fullyOrderedSymbolsByKind)
+                            {
+                                var symbolType = symbol.GetSymbolType();
+
+                                if (semanticModel.Compilation.ClassifyCommonConversion(symbolType, targetType).IsImplicit)
+                                {
+                                    callsiteInferenceFailed = false;
+
+                                    if (symbol.IsKind(SymbolKind.Method))
+                                    {
+                                        expression = Generator.InvocationExpression(Generator.IdentifierName(symbol.Name));
+                                    }
+                                    else
+                                    {
+                                        expression = Generator.IdentifierName(symbol.Name);
+                                    }
+
+                                    break;
+                                }
+                            }
+                        }
+
+                        if (expression == null)
+                        {
+                            expression = Generator.ParseExpression((isCallsiteActuallyErrored || callsiteInferenceFailed) ? "TODO" : addedParameter.CallSiteValue);
```

Comment:
what is isCallsiteActuallyErrored ?

Reply:
I've renamed it to `isCallsiteActuallyTODO`. Does that help?